vfs/dir.go

```diff
@@ -371,7 +371,9 @@ func (d *Dir) renameTree(dirPath string) {
 	// Make sure the path is correct for each node
 	if d.path != dirPath {
 		fs.Debugf(d.path, "Renaming to %q", dirPath)
+		delete(d.parent.items, name(d.path))
```

Comment:
This is a good fix, but it really needs a test. I try to keep the VFS coverage up as all these corner cases are really difficult to track down!

Reply:
Ok, I'll try to add one! (Although FWIW, I found this because it was causing `TestSFTPRclone:` to fail!)

Reply:
Also btw, I am now seeing a bunch of `expecting X to have MkdirMetadata method: optional feature not implemented` and `expecting X to have SetMetadata method` errors that I'm pretty sure were not there before...maybe related to the recent `--create-empty-src-dirs` change?

https://pub.rclone.org/integration-tests/2024-05-13-010010/sftp-cmd.bisync-TestSFTPRclone-5.txt

Reply:
I added a test.

Reply:
Thank you. I've cherry picked this :-)